java-squid/src/main/java/org/sonar/java/cfg/CFG.java

```diff
@@ -0,0 +1,757 @@
+        buildMemberSelect((MemberSelectExpressionTree) tree);
+        break;
+      case CONDITIONAL_AND:
+        buildConditionalAnd((BinaryExpressionTree) tree);
+        break;
+      case CONDITIONAL_OR:
+        buildConditionalOr((BinaryExpressionTree) tree);
+        break;
+      case LABELED_STATEMENT:
+        buildLabeledStatement((LabeledStatementTree) tree);
+        break;
+      case SWITCH_STATEMENT:
+        buildSwitchStatement((SwitchStatementTree) tree);
+        break;
+      case BREAK_STATEMENT:
+        buildBreakStatement((BreakStatementTree) tree);
+        break;
+      case CONTINUE_STATEMENT:
+        buildContinueStatement((ContinueStatementTree) tree);
+        break;
+      case WHILE_STATEMENT:
+        buildWhileStatement((WhileStatementTree) tree);
+        break;
+      case DO_STATEMENT:
+        buildDoWhileStatement((DoWhileStatementTree) tree);
+        break;
+      case FOR_EACH_STATEMENT:
+        buildForEachStatement((ForEachStatement) tree);
+        break;
+      case FOR_STATEMENT:
+        buildForStatement((ForStatementTree) tree);
+        break;
+      case TRY_STATEMENT:
+        buildTryStatement((TryStatementTree) tree);
+        break;
+      case THROW_STATEMENT:
+        buildThrowStatement((ThrowStatementTree) tree);
+        break;
+      case SYNCHRONIZED_STATEMENT:
+        buildSynchronizedStatement((SynchronizedStatementTree) tree);
+        break;
+      case POSTFIX_INCREMENT:
+      case POSTFIX_DECREMENT:
+      case PREFIX_INCREMENT:
+      case PREFIX_DECREMENT:
+      case UNARY_MINUS:
+      case UNARY_PLUS:
+      case BITWISE_COMPLEMENT:
+      case LOGICAL_COMPLEMENT:
+        buildUnaryExpression((UnaryExpressionTree) tree);
+        break;
+      case PARENTHESIZED_EXPRESSION:
+        build(((ParenthesizedTree) tree).expression());
+        break;
+      case ARRAY_ACCESS_EXPRESSION:
+        buildArrayAccessExpression((ArrayAccessExpressionTree) tree);
+        break;
+      case ARRAY_DIMENSION:
+        buildArrayDimension((ArrayDimensionTree) tree);
+        break;
+      case NEW_CLASS:
+        buildNewClass((NewClassTree) tree);
+        break;
+      case IDENTIFIER:
+      case INT_LITERAL:
+      case LONG_LITERAL:
+      case DOUBLE_LITERAL:
+      case CHAR_LITERAL:
+      case FLOAT_LITERAL:
+      case STRING_LITERAL:
+      case BOOLEAN_LITERAL:
+      case NULL_LITERAL:
+        currentBlock.elements.add(tree);
+        break;
+      case TYPE_CAST:
+        buildTypeCast(tree);
+        break;
+      case INSTANCE_OF:
+        buildInstanceOf((InstanceOfTree) tree);
+        break;
+      case NEW_ARRAY:
+        buildNewArray((NewArrayTree) tree);
+        break;
+      case CLASS:
+      case ENUM:
+      case ANNOTATION_TYPE:
+      case INTERFACE:
+        // Ignore declarations
+      case METHOD_REFERENCE:
+        // Java 8 constructions : ignored for now.
+      case EMPTY_STATEMENT:
+        break;
+      case LAMBDA_EXPRESSION:
```

Comment:
![MAJOR](https://raw.githubusercontent.com/SonarCommunity/sonar-github/master/images/severity-major.png) This case's code block is the same as the block for the case on line 304. [![rule](https://raw.githubusercontent.com/SonarCommunity/sonar-github/master/images/rule.png)](http://nemo.sonarqube.org/coding_rules#rule_key=squid%3AS1871)